utils/wasm-builder/src/wasm_project.rs

```diff
@@ -229,6 +229,7 @@ fn create_project_cargo_toml(
 	let mut release_profile = Table::new();
 	release_profile.insert("panic".into(), "abort".into());
 	release_profile.insert("lto".into(), true.into());
+	release_profile.insert("codegen-units".into(), 1.into());
```

Comment:
Do you think it makes sense to have a `production` profile here as to reduce compile time for the normal dev-flow?  
Polkadot opted for it, and Substrate in general as well.

Reply:
Yeah this would make sense. This is just an ad-hoc commit that will not be merged. However, it is already build with lto for every release build. Do you think `cgu=1` is so much slower that it is worth an extra profile?

Reply:
I did some benchmarks in https://github.com/paritytech/substrate/issues/1262, it does make things significantly slower (tested `node-runtime` and Polkadot compilation there).

Also note that `release_profile.insert("lto".into(), true.into());` is `-C lto`, not `-C lto=fat`!

Reply:
`to=true` and `lto="fat"` are the same: https://doc.rust-lang.org/cargo/reference/profiles.html#lto

Reply:
Hm, it does say it is the same, but the result is that the resulting artifact size is different, so it must be different.

Reply:
Couldn't it be something else? Do you have the same size when you compile back2back with the same flags?

Reply:
I reproducibly get 2 different sizes when switching between `-C lto` and `-C lto=fat`. Created zulip thread about this: https://rust-lang.zulipchat.com/#narrow/stream/122652-new-members/topic/-C.20lto.20and.20-C.20lto-fat.20are.20different.3F

Reply:
I just tested `wasmi` with `lto="fat"` and `lto=true` and I received the same outputs everytime. In some cases Cargo did not even bother to recompile the crate. So I cannot approve what you are experiencing unfortunately.

Reply:
I was adding `.append_to_rust_flags("-C lto=fat")` to `build.rs`, I guess you changed those lines above instead? Not sure how there could be any difference though.

Reply:
Yes I just changed the lines that @athei also changed.

Reply:
If you append it to the build flags in the `build.rs` you will have essentially both: `true` in the profile (set by the `substrate-wasm-builder` as per my patch) and then the hard coded flags (as set by you). Not sure why this makes a difference.